### PR TITLE
Apply expected file permissions to linux files after authenticode signing

### LINKED ISSUE #18495: Incorrect file permissions in latest release https://github.com/PowerShell/PowerShell/releases/tag/v7.3.0
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

tarfiles contain incorrect file permissions. the executables are makred as rw and NOT executable. not an issue on previous releases (eg: they are executable on 7.2.7). this also affects nearly all other files in the tarfile (most of them have different permissions from before)

### Expected behavior

executables are marked as executable
```console
-rwxr-xr-x
```

### Actual behavior

executables are not marked as executable and many files have different permissions
```console
-rw-r--r--
```

### Error details

_No response_

### Environment data

```powershell
Linux
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

When trying to authenticode sign files on Windows the linux file permissions are lost. We store them in a json file and then re-apply them.

Fixes https://github.com/PowerShell/PowerShell/issues/18495

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -4623,6 +4644,76 @@ function Invoke-AzDevOpsLinuxPackageBuild {
     }
 }
 
+function Set-LinuxFilePermission {
```
Comment (TravisEz13) [inline tools/packaging/packaging.psm1:4647]: At least some minimal description of the function

**Review by TravisEz13 — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]
LGTM

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (adityapatwardhan) [reply tools/packaging/packaging.psm1:4663]: Added comments

**Pull request merged**